Description: Output stdout and stderr if system test fails
  To make it easy to debug a failing system_test.sh run on any CI system
  output the standard output and error contents of the previous command.
  .
  Also set default git name/email to avoid unnecassary git warnings in the
  CI logs.
Author: Otto Kekäläinen <otto@debian.org>
Forwarded: no
--- a/system_test.sh
+++ b/system_test.sh
@@ -22,7 +22,16 @@ function try { let tests+=1; this="$1";
 
 function assert {
 	[[ "$1" == "$2" ]] && { printf "."; return; }
-	printf "\nFAIL: $this\n'$1' != '$2'\n"; exit 1
+	printf "\nFAIL: $this\n'$1' != '$2'\n"
+
+	echo "*************************************"
+	echo "Preceding stdout:"
+	cat $tmp/exec.out
+	echo "*************************************"
+	echo "Preceding stderr:"
+  cat $tmp/exec.err
+	echo "*************************************"
+  exit 1
 }
 function skip { printf "s"; }
 
@@ -37,6 +46,18 @@ clear_tmp='rm -rf $tmp'
 trap "$clear_tty; $clear_tmux; $clear_tmp" EXIT
 trap 'printf "\nTerminated by SIGINT at line $LINENO\n"; exit 1' INT
 
+# required binary
+
+ENTR="$(./entr 2>&1)"
+if echo "$ENTR" | grep -qF release
+then
+	echo "Testing Entr " "$(echo "$ENTR" | grep -F release)"
+else
+	echo $ENTR
+	echo "Entr binary did not run as expected"
+	exit 1
+fi
+
 # required utilities
 
 utils="file pgrep git vim tmux"
